File: javascript/prime-factors/prime-factors.js
var primeFactors = function() {};

primeFactors.prototype.for = function(number) {
	var factors = [];
	
	for (var i = 2; i <= number; ++i) {
		if (number % i === 0) {
			number = number / i;
			factors.push(i);
			
			i = 1;
		}
	}
	
	return factors;
};

module.exports = new primeFactors();

File: python/prime-factors/prime_factors.py
def prime_factors(num):
    factors = []
    
    i = 2
    while (i <= num):
        if 0 == (num % i):
            factors.append(i)
            
            # NOTE: This hides an implicit cast to int
            num = num // i
            i = 1
        
        i += 1
    
    return factors